source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -18,20 +19,25 @@ function isChecked(file: HTMLElement): boolean {
 }
 
 function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
-	if (!event.shiftKey || !previousFile) {
+	if (!event.shiftKey) {
+		return;
+	}
+
+	const isFirstToggle = previousFile === undefined;
+
+	previousFile ??= select('.js-file'); // #5484
```

Comment:
This should now be unnecessary. This "if missing" is already part of `getItemsBetween`

`previousFileState` should be dropped as suggested, we should only follow the current file state. It may be best to fix the whole behavior in this PR I think